redisgears_core/src/lib.rs

```diff
@@ -573,6 +578,103 @@ pub(crate) fn call_redis_command(
     ctx.call_ext(command, call_options, args)
 }
 
+/// a struct that holds information about not yet resolve future replies.
+/// The struct allows to either abort the execution or invoke the on done callback.
+struct FutureHandlerContext {
+    callback: Option<Box<dyn FnOnce(&Context, CallResult<'static>)>>,
```

Comment:
Its also an option so we will call it once because there is a chance that the abort fails and then it will call the unblock handler.

Reply:
Yes this is why we need the option, otherwise we would not be able to call it at all using the mut ref. And if we change the interface to get self then we would not be able to call it from the share reference that is held both by the unblock handler and by the global future_handlers vector (that is used to abort it on failovers).